src/RA_User.cpp

```diff
@@ -86,36 +110,17 @@ void LocalRAUser::AttemptLogin(bool bBlocking)
         DialogBox(g_hThisDLLInst, MAKEINTRESOURCE(IDD_RA_LOGIN), g_RAMainWnd, RA_Dlg_Login::RA_Dlg_LoginProc);
         ra::services::ServiceLocator::Get<ra::services::IConfiguration>().Save();
     }
-
 }
 
 void LocalRAUser::AttemptSilentLogin()
 {
-    //	NB. Don't login here: cause a login when http request returns!
-    PostArgs args;
-    args['u'] = Username();
-    args['t'] = Token();
-    RAWeb::CreateThreadedHTTPRequest(RequestLogin, args);
-}
-
-void LocalRAUser::HandleSilentLoginResponse(rapidjson::Document& doc)
-{
-    if (doc.HasMember("Success") && doc["Success"].GetBool())
-    {
-        const std::string& sUser = doc["User"].GetString();
-        const std::string& sToken = doc["Token"].GetString();
-        const unsigned int nPoints = doc["Score"].GetUint();
-        const unsigned int nUnreadMessages = doc["Messages"].GetUint();
-        ProcessSuccessfulLogin(sUser, sToken, nPoints, nUnreadMessages, TRUE);
-    }
-    else if (doc.HasMember("Error"))
+    ra::api::Login::Request request;
+    request.Username = Username();
+    request.ApiToken = Token();
+    request.CallAsync([this](const ra::api::Login::Response& response)
     {
-        MessageBox(nullptr, NativeStr(doc["Error"].GetString()).c_str(), TEXT("Login Failed"), MB_OK);
-    }
-    else
-    {
-        MessageBox(nullptr, TEXT("Login failed, please login again."), TEXT("Login Failed"), MB_OK);
-    }
```

Comment:
Parsing of response and error handling was largely duplicated in the login dialog. Now that's all in the `ConnectedServer` helper functions.